### PR TITLE
Document what lives in contrib

### LINKED ISSUE #325: Document what's in contrib
The non-essential bits that are isolated to the contrib directory aren't all that well documented, imo. Questions like [this one](https://github.com/kubernetes-incubator/service-catalog/issues/316) are coming in, for instance, and I have realized I don't even know how to respond... because I don't have a clear grasp of everything that's in contrib.

For each major component in contrib, e.g. brokers, registry, etc., can someone in the know add some documentation to explain what it is, what value it adds, etc.

In some cases, are some of these things not directly serving our collective goals anymore? It's not clear.

### PR DESCRIPTION
**Pull request by pmorie**
Closes #325

### CODE REVIEW

**Comment by pmorie** [comment]
@krancour does this meet your expectation?

File: docs/devguide.md [hunk 1]
```diff
@@ -25,7 +25,12 @@ layout:
     │   └── apiserver           # The service catalog API server binary
     │   └── controller-manager  # The service catalog controller manager binary
     ├── contrib                 # Contains all non-essential source
+    │   └── build               # Dockerfiles for contrib images (example: ups-broker)
+    │   └── cmd                 # Entrypoints for contrib binaries
+    │   └── examples            # Example API resources
     │   └── hack                # Non-build related scripts
+    │   └── jenkins             # Jenkins configuration
+    │   └── pkg                 # Contrib code
```
Comment (MHBauer) [inline docs/devguide.md]: Is it obvious that this is go code only?

File: docs/devguide.md [hunk 2]
```diff
@@ -25,7 +25,12 @@ layout:
     │   └── apiserver           # The service catalog API server binary
     │   └── controller-manager  # The service catalog controller manager binary
     ├── contrib                 # Contains all non-essential source
```
Comment (simonleung8) [inline docs/devguide.md]: When I first read `non-essential` source, I though of 'old, stale or wip' code from the service catalog project. It was only at a later time I realized it has sample code and script files for builds among others.

Comment (pmorie) [inline docs/devguide.md]: Do you have a suggestion for another name?

Comment (simonleung8) [inline docs/devguide.md]: What do you think if we just spell out what is in there. For example, "build scripts, sample or mock code for broker..."

**Comment by pmorie** [comment]
@MHBauer @simonleung8 feedbacks addressed

**Pull request merged**